ts/src/main/scala/ch/epfl/ts/example/FullMarketSimulationWithCustomSetup.scala

```diff
@@ -0,0 +1,220 @@
+      MadTrader.ORDER_VOLUME -> NaturalNumberParameter(10),
+      MadTrader.CURRENCY_PAIR -> new CurrencyPairParameter(Currency.EUR, Currency.CHF))
+
+    val tId = 15L
+    val marketId = MarketNames.FOREX_ID
+
+    val trader = MadTrader.getInstance(tId, List(marketId), parameters, "OneMadTrader")
+    addConsument(classOf[Quote], trader)
+    addConsument(classOf[TheTimeIs], trader)
+    val broker = builder.createRef(Props(classOf[StandardBroker]), "Broker")
+    addConsument(classOf[Quote], broker)
+
+    trader->(broker, classOf[Register])
+    trader->(broker, classOf[FundWallet])
+    connectAllOrders(trader, broker)
+    
+    // MarketMaker
+    val parameters2 = new StrategyParameters(
+      MarketMakerTrader.INITIAL_FUNDS -> WalletParameter(Map(Currency.CHF -> 10000.0, Currency.EUR -> 10000.0)),
+      MarketMakerTrader.INTERVAL -> new TimeParameter(1 seconds),
+      MarketMakerTrader.ORDER_VOLUME -> NaturalNumberParameter(10),
+      MarketMakerTrader.CURRENCY_PAIR -> new CurrencyPairParameter(Currency.EUR, Currency.CHF))
+    
+    val marketMakerParameters = new StrategyParameters(
+        MMwithWallet.INITIAL_FUNDS -> WalletParameter(Map(Currency.CHF -> 10000.0, Currency.EUR -> 10000.0)),
+        MMwithWallet.SYMBOL -> new CurrencyPairParameter(Currency.EUR, Currency.CHF),
+        MMwithWallet.SPREAD -> new RealNumberParameter(0.001))
+
+    val tId2 = 100L
+
+//    val marketMaker = MarketMakerTrader.getInstance(tId2, List(marketId), parameters2, "OneMarketMakerTrader")
+    val marketMaker = MMwithWallet.getInstance(tId2, List(marketId), marketMakerParameters, "WalletMarketMakerTrader")
+    addConsument(classOf[Quote], marketMaker)
+    addConsument(classOf[TheTimeIs], marketMaker)
+
+    marketMaker->(broker, classOf[Register])
+    marketMaker->(broker, classOf[FundWallet])
+    connectAllOrders(marketMaker, broker)
+
+    // Fetcher
+    val fetcher = createFetcher(useLiveData, builder, symbol)
+    
+    // Hybrid market
+    val fetcherRules = new FxMarketRulesWrapper
+    val simulationRules = new SimulationMarketRulesWrapper
+    val market = builder.createRef(Props(classOf[HybridMarketSimulator], marketId, fetcherRules, simulationRules), MarketNames.FOREX_NAME)
+    fetcher->(market, classOf[Quote])
+
+    addProducer(classOf[Quote], market)
+    addProducer(classOf[TheTimeIs], market)
+    connectAllOrders(broker, market)
+    market->(broker, classOf[ExecutedBidOrder], classOf[ExecutedAskOrder])
+    
+    // for the market maker
+//    connectAllOrders(marketMaker, market) // TODO use broker inbetween
+    market->(marketMaker, classOf[MarketBidsEmpty], classOf[MarketAsksEmpty], classOf[MarketEmpty], classOf[ExecutedBidOrder], classOf[ExecutedBidOrder])
+
+    connectProducersWithConsuments()
+    
+    builder.start
+    val delay = 20 * 1000 //in ms
+    scheduleChange(fetcher, market, delay)
+  }
+
+  def scheduleChange(quoteFetcher: ComponentRef, market: ComponentRef, delay: Long) = {
```

Comment:
Do not duplicate all these functions (can easily be avoided by factoring those out to a super-class dedicated to Full Market Simulation examples)